core/src/main/scala/io/qbeast/core/model/AutoIndexer.scala

```diff
@@ -0,0 +1,36 @@
+package io.qbeast.core.model
+
+/**
+ * AutoIndexer interface to automatically choose which columns to index.
+ * @tparam DATA
+ *   the data to index
+ */
+trait AutoIndexer[DATA] {
```

Comment:
The name is not what excites me. IMHO it sounds like it is responsible for indexing, that is not the case. However since I cannot provide anything better than `ColumnsToIndexChooser`, you can leave it as is.

Reply:
This class is pretty technical, so IMHO the name can follow the name of its main member. Currently its duty is to `chooseColumsToIndex`, the verb is "choose", thus we can derive a noun for the class i.e. `...Chooser` (Chooser is a quite used word in software, for example FileChooser in UI). Since it chooses columns to index, I put these words to the beginning. This is how I got `ColumnsToIndexChooser`.
If you prefer "select", then same logic leads us to `ColumnsToIndexSelector`. I believe that it is a common practice to move the objective before the noun derived from the verb representing the main operation. At least for small technical classes which do just one thing. For bigger entities like services with many operations it is a bit different.

Reply:
Yep, I understand. In my opinion the word `select` feels more elegant. I still do not like the whole composition as `ColumnsToIndexSelector`, but it is as descriptive as it can be. 

For the same reasoning argument, I would like to also change the configuration parameters. 

**Which one feels best?**

```scala
spark.qbeast.index.columnsToIndex.auto=true
spark.qbeast.index.columnsToIndex.max=10
```
 or

```scala
spark.qbeast.index.columnSelector.auto=true
spark.qbeast.index.columnSelector.max=10
```

or 

```scala
spark.qbeast.index.columnsToIndexSelector.auto=true
spark.qbeast.index.columnsToIndexSelector.max=10
```

Reply:
I think the first option is the best. Because they are for the end user, who knows that the columsToIndex can be chosen automatically. So columnsToIndex.auto is good. Unfortunately columnsToIndex.max is misleading, because one can think that it is the maximum number of columns to specified by columnsToIndex option.

Reply:
The following variants look good to me:
```
spark.qbeast.index.columnsToIndex.auto=true
spark.qbeast.index.columnsToIndex.auto.max=10
```
or
```
spark.qbeast.index.columnsToIndex.auto.enabled=true
spark.qbeast.index.columnsToIndex.auto.max=10
```
I would go with the first of the two above for brevity.

Reply:
These options are for the end user at least in theory, so they should be named in terms the user can observe and understand. So the variants with "selector" expose and refer to the name of the internal abstraction which is not visible to the user.

Reply:
Those seems appropiate: 
```
spark.qbeast.index.columnsToIndex.auto=true
spark.qbeast.index.columnsToIndex.auto.max=10
```

Ok, thanks!